cartridges/bm_affirm/cartridge/scripts/handleTransaction.js

```diff
@@ -0,0 +1,274 @@
+			case 'partially_refunded':
+				customStatus = 'PARTIALLY_CAPTURED'
+				break
+			case 'captured':
+				customStatus = 'CAPTURE'
+				break
+			case 'refunded':
+				customStatus = 'REFUNDED'
+				break
+			default:
+				log.error("Invalid Affirm Status returned")
+				return {
+					status: false,
+					error: true,
+				}
+		}
+
+		Transaction.wrap(function () {
+			if (Order.custom.AffirmStatus.value !== customStatus) {
+				Order.custom.AffirmStatus = customStatus
+			}
+			Order.custom.AffirmTransactionHistory = JSON.stringify(events)
+		})
+		return {
+			status: true,
+			error: false,
+		}
+	} catch (e) {
+		log.error("Exception occurred: " + e)
+		return {
+			status: false,
+			error: true,
+		}
+	}
+}
+
+/**
+ * Cancel (Void) action
+ * @param {string} transactionID - Affirm Transaction ID
+ * @returns {Object} response
+ */
+function cancel(transactionID) {
+	var Order = OrderMgr.searchOrder(
+		"custom.AffirmExternalId = {0}",
+		transactionID
+	)
+	var authAmount = parseFloat(Order.custom.AffirmAuthAmount)
+
+	if (Order.custom.AffirmStatus.value !== "AUTH") {
+		log.error("Invalid void action")
+		return {
+			status: false,
+			error: true,
+		}
+	}
+
+	try {
+		var _r = api.void(transactionID)
+		Transaction.wrap(function () {
+			Order.custom.AffirmStatus = "VOIDED"
+			Order.setPaymentStatus(Order.PAYMENT_STATUS_NOTPAID)
+			Order.setStatus(Order.ORDER_STATUS_CANCELLED)
+		})
+		return {
+			status: true,
+			error: false,
+		}
+	} catch (e) {
+		log.error("Exception occurred: " + e)
+		return {
+			status: false,
+			error: true,
+		}
+	}
+}
+
+/**
+ * Capture action
+ * @param {string} transactionID - transaction id
+ * @param {string} amount - capture amount
+ * @returns {Object} transaction details
+ */
+function capture(transactionID, amount) {
+	var Order = OrderMgr.searchOrder(
+		"custom.AffirmExternalId = {0}",
+		transactionID
+	)
+	var status = false
+	var error
+	var amountToCapture = parseFloat(amount)
+	var authAmount = parseFloat(Order.custom.AffirmAuthAmount)
+	var capturedAmount = parseFloat(Order.custom.AffirmCapturedAmount)
+
+	switch (Order.custom.AffirmStatus.value) {
+		case "AUTH":
+		case "PARTIALLY_CAPTURED":
+			break
+		default:
+			log.error("Invalid capture action")
+			return {
+				status: false,
+				error: true,
+			}
+	}
+
+	if (
+		typeof amountToCapture == "undefined" ||
+		(typeof amountToCapture !== "undefined" && amountToCapture <= 0)
+	) {
+		log.error("Invalid capture amount")
+		return {
+			status: false,
+			error: true,
+		}
+	}
+
+	try {
+		var _r = api.capture(transactionID, Order.orderNo, amountToCapture)
+		var newCapturedAmount = capturedAmount + parseFloat(_r.response.amount)
+		Transaction.wrap(function () {
+			Order.custom.AffirmCapturedAmount = newCapturedAmount
+			if (authAmount === newCapturedAmount) {
+				Order.custom.AffirmStatus = "CAPTURE"
+				Order.setPaymentStatus(Order.PAYMENT_STATUS_PAID)
+				Order.setStatus(Order.ORDER_STATUS_COMPLETED)
+			} else {
```

Comment:
where do we check if the new captured amount request total is greater than auth amount?

Reply:
@nandadubey should we handle the condition where cap amount is larger than auth amount? Since we don't allow over capture I couldn't think of the use case for such condition

Reply:
No i meant where are we checking that it's a over capture and throw an error, does that get handled somewhere else?

Reply:
We checked the capture amount input (< remaining auth amount) here: 
https://github.com/Affirm/Salesforce_Commerce_Cloud_Affirm/blob/feature_split-capture/cartridges/bm_affirm/cartridge/static/default/js/transactions.js#L86